hydroflow/tests/lattice_join.rs

```diff
@@ -0,0 +1,134 @@
+use hydroflow::util::collect_ready;
+use hydroflow_macro::hydroflow_syntax;
+use multiplatform_test::multiplatform_test;
+
+#[multiplatform_test]
+pub fn test_lattice_join_reducing_behavior() {
+    // lattice_fold has this particular behavior where it uses LatticeFrom to convert the first element into the another lattice type and then it folds into that accumulator.
```

Comment:
nits: 
- typo: "the another". 
- "this particular behavior where" -> "the following particular initialization behavior:". It's a bit ambiguous what "this" is a reference to, i had to read it twice.